packages/grafana-data/src/types/data.ts

```diff
@@ -17,13 +17,24 @@ export enum LoadingState {
 
 export type PreferredVisualisationType = 'graph' | 'table' | 'logs' | 'trace';
 
+export enum FieldNamingStrategy {
+  Smart = 'smart',
+  Complete = 'complete',
+  FrameName = 'frame-name',
+}
+
 export interface QueryResultMeta {
   /** DatasSource Specific Values */
   custom?: Record<string, any>;
 
   /** Stats */
   stats?: QueryResultMetaStat[];
 
+  /**
+   * Field naming strategy
+   */
+  namingStrategy?: FieldNamingStrategy;
```

Comment:
Open question is should this live on field instead? we do not have field meta, so would be on FieldConfig

Reply:
and importantly, what happens after a transform.  For example if a prometheus field wants the name from the frame... what happens after a transform?  do we make sure to render it first?

Reply:
the "frame-name" strategy really only makes sense with time/value frames (simple timeseries)

Reply:
yea, agree so this naming strategy only makes sense for time series pre-transform, which is why I think it can live on frame. but the other two (short/smart & complete) makes sense on field, so maybe this is a field config option.